frontend/sw360-portlet/src/main/java/com/siemens/sw360/portal/portlets/licenses/LicensesPortlet.java

```diff
@@ -155,6 +197,81 @@ private void addLicenseBreadcrumb(RenderRequest request, RenderResponse response
+        LicenseService.Iface client = thriftClients.makeLicenseClient();
+        String licenseId = request.getParameter(LICENSE_ID);
+        User user = UserCacheHolder.getUserFromRequest(request);
+
+        if (!Strings.isNullOrEmpty(licenseId)) {
+            try {
+                license = client.getByID(licenseId, user.getDepartment());
+            } catch (TException e) {
+                log.error("Error updating license:", e);
+            }
+        }
+
+        license = updateLicenseFromRequest(license,request);
+
+        RequestStatus requestStatus=RequestStatus.FAILURE;
+        try {
+            requestStatus = client.updateLicense(license, user, user);
+        } catch (TException e) {
+            log.error("Could not add or update license:" + e);
+        }
+        if (licenseId != null) {
+            response.setRenderParameter(LICENSE_ID, licenseId);
+            response.setRenderParameter(PAGENAME, PAGENAME_DETAIL);
+            response.setRenderParameter(SELECTED_TAB, "Details");
+            setSessionMessage(request, requestStatus, "License", "update");
+        } else {
+            response.setRenderParameter(PAGENAME, PAGENAME_VIEW);
+            setSessionMessage(request, requestStatus, "License", "adde");
+        }
+    }
+
+    private License updateLicenseFromRequest(License license, ActionRequest request) {
+        String text = request.getParameter(License._Fields.TEXT.name());
+        String fullname = request.getParameter(License._Fields.FULLNAME.name());
+        String shortname = request.getParameter(License._Fields.SHORTNAME.name());
+        boolean gpl2compatibility =
+                (request.getParameter(License._Fields.GPLV2_COMPAT.toString()) == null) ? false : true;
+        boolean gpl3compatibility =
+                (request.getParameter(License._Fields.GPLV3_COMPAT.toString()) == null) ? false : true;
+        String licenseTypeString =
+                request.getParameter(License._Fields.LICENSE_TYPE.toString() + LicenseType._Fields.LICENSE_TYPE.toString());
+        license.setText(CommonUtils.nullToEmptyString(text));
+        license.setFullname(CommonUtils.nullToEmptyString(fullname));
+        license.setShortname((CommonUtils.nullToEmptyString(shortname)));
+        license.setGPLv2Compat(gpl2compatibility);
+        license.setGPLv3Compat(gpl3compatibility);
+        try {
+            String licenseTypeDatabaseId = getDatabaseIdFromLicenseType(licenseTypeString);
+            license.setLicenseTypeDatabaseId(licenseTypeDatabaseId);
+        } catch (TException e) {
+            log.error("Could not set licenseTypeDatabaseId:" + e);
+        }
+        return license;
+    }
+
+    private String getDatabaseIdFromLicenseType(String licenseTypeString) throws TException {
+        if (licenseTypes == null) {
+            LicenseService.Iface client = thriftClients.makeLicenseClient();
+            try {
+                licenseTypes = client.getLicenseTypes();
```

Comment:
Because I could have got it from the db also in prepareEditView. If this is the case I don't have to fetch it a second time from the db.


Von meinem Samsung Gerät gesendet.

-------- Ursprüngliche Nachricht --------
Von: Maximilian Huber <notifications@github.com> 
Datum: 22.02.2016  13:25  (GMT+01:00) 
An: siemens/sw360portal <sw360portal@noreply.github.com> 
Cc: heydenreich <birgit.heydenreich@tngtech.com> 
Betreff: Re: [sw360portal] Edit licenses (#94) 

> +        license.setGPLv2Compat(gpl2compatibility);
> +        license.setGPLv3Compat(gpl3compatibility);
> +        try {
> +            String licenseTypeDatabaseId = getDatabaseIdFromLicenseType(licenseTypeString);
> +            license.setLicenseTypeDatabaseId(licenseTypeDatabaseId);
> +        } catch (TException e) {
> +            log.error("Could not set licenseTypeDatabaseId:" + e);
> +        }
> +        return license;
> +    }
> +
> +    private String getDatabaseIdFromLicenseType(String licenseTypeString) throws TException {
> +        if (licenseTypes == null) {
> +            LicenseService.Iface client = thriftClients.makeLicenseClient();
> +            try {
> +                licenseTypes = client.getLicenseTypes();

Why do you save the list of `licenseTypes` in a field?

---
Reply to this email directly or view it on GitHub:
https://github.com/siemens/sw360portal/pull/94/files#r53617717